src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +56,360 @@ uint32_t SystemRequest::index = 0;
 const std::string kSYNC = "SYNC";
 const std::string kIVSU = "IVSU";
 
+static const char* kQeeryAppsValidationFailedPrefix =
+    ":QUERY_APPS_VALIDATION_FAILED: ";
+
+template <typename ObjectType>
+class QueryAppsDataValidatorTemplate {
+ public:
+  typedef std::set<std::string> SynonymsSetType;
+  typedef std::map<std::string, SynonymsSetType> SynonymsMapType;
+
+  QueryAppsDataValidatorTemplate(const ObjectType& o)
```

Comment:
@okoreniak We have to avoid using such unclear names like `o`. But in this particular case may be it's normal. I'm not sure.